src/front/domain_socket.rs

```diff
@@ -202,11 +203,10 @@ impl Listen for DomainSocketListener {
                     format_error!("Failed to set stream as blocking", err);
                     None
                 } else {
+                    let metadata = stream.metadata();
```

Comment:
Is the `GetMetadata` trait needed here if we are already in Unix specific code? As in, could you just replace the implementation of that trait on `UnixStream` here?

Reply:
Just pinging this question as I think it is still valid

Reply:
We could -- this was more just thinking for the future. Will revert this, apologies for the lag on responding to this comment. 😄